### DISCUSSION TITLE
How to Conect to RESTful Interface?

### DISCUSSION THREAD

**Discussion by bcgordon3** [zauberzeug/nicegui#447] [Q&A] [answered]
Hello,
I'm trying to use NiceGUI to create a quick frontend to help load/unload items to a restful API. I looked at the example for fastapi here
https://github.com/zauberzeug/nicegui/tree/main/examples/fastapi
But it looks like that is running a fastapi app on python connected to the frontend on the same machine and passing that . That is not be the case for my scenario. I'm fairly new to the webpage side of things so I may be asking something where it should be painfully obvious. The restful api will be at <ip>:<port>/status or /config or whatnot with JSON responses to requests. What would be the right way to hook into that api?

**Comment by rodja** [comment, marked as answer]
NiceGUI is simply a frontend to any Python code you like. So you can implement the REST calls on the server and only present the options and results with NiceGUI. It's advisable to use an async http library like [httpx](https://www.python-httpx.org/async/) to not hinder the responsiveness of the user interface.

**Reply by bcgordon3** [reply]
Ok, gotcha. I'm trying to make it do more than it was designed. I'll work the API using normal python and nicegui as the display.
